### DISCUSSION TITLE
Do not `set` `$PATH` in `config.fish`, suggest using `fish_user_paths` instead

### DISCUSSION THREAD

**Discussion by catthehacker** [Homebrew/discussions#126] [Everyday usage] [answered]
Currently Homebrew is appending `$PATH` via https://github.com/Homebrew/brew/commit/871974650b16d9d4b741fec8986608b081e294a4
Instead it should not do it at all and suggest user to run this command once
```fish
set -U fish_user_paths "/usr/local/sbin" $fish_user_paths
```
The reasoning is that `config.fish` is universal config shared across machines. `$PATH` set in `config.fish` will follow to any other host.
`fish_user_paths` is a variable set only for the current host via `fishd.*` so it will not bloat `$PATH` on other hosts.
[Also explained here...](https://github.com/Homebrew/brew/issues/8965#issuecomment-724705838)

**Comment by SMillerDev** [comment, marked as answer]
I'm not really sure I understand your reasoning about sharing your dotfiles between machines. But if you think something in homebrew could be better you should make a pull request to fix it.

**Reply by catthehacker** [reply]
The reason is that I can have unified configuration across machines, like prompt or some functions :) 
![image](https://user-images.githubusercontent.com/31106839/98690973-d2cd2800-236d-11eb-8a68-32ed1c471b7e.png) *don't judge*
> But if you think something in homebrew could be better you should make a pull request to fix it.

I wanted to know maintainers opinion before I start doing anything, as sometimes some of the developers are *stubborn*, don't see the benefit and close the PR or just simply ignore it.
I'll make PR once I go through whole code where it mentions `fish`shell.

**Reply by SMillerDev** [reply]
> The reason is that I can have unified configuration across machines, like prompt or some functions :)

But there's no functional difference between PATH and the fish equivalent for it as far as I can see.

**Reply by catthehacker** [reply]
Whichever option is chosen, path(s) always end up in `$PATH` but its up to the implementation how that will work and what are the side effects.
[From `fish` documentation:](https://fishshell.com/docs/current/tutorial.html#path)
> A faster way is to modify the `$fish_user_paths` universal variable, which is automatically prepended to `$PATH`. For example, to permanently add `/usr/local/bin` to your `$PATH`, you could write:
> ```fish
> set -U fish_user_paths /usr/local/bin $fish_user_paths
> ```
> The advantage is that you don't have to go mucking around in files: just run this once at the command line, and it will affect the current session and all future instances too. (Note: you should NOT add this line to `config.fish`. If you do, the variable will get longer each time you run fish!)

**Reply by SMillerDev** [reply]
They don't actually advise against using the PATH variable, I think we should keep the current approach since it's easier for people to understand. As a fish user that syncs their dotfiles between devices I don't even understand what the benefit of `fish_user_paths` is in this case, other than "it's fish specific"

**Reply by catthehacker** [reply]
I'm not saying it's bad, just that there is a better way to do it.
> As a fish user that syncs their dotfiles between devices I don't even understand what the benefit of `fish_user_paths` is in this case, other than "it's fish specific"

The benefit is that `$fish_user_paths` is changing `$PATH` only on machine it was set on via `fishd.*` file, while `$PATH` set via `config.fish` is affecting `$PATH` on all machines.
> I think we should keep the current approach since it's easier for people to understand.

That's okay, it's the reason why I wanted to discuss it before. :)
